### DISCUSSION TITLE
need json schema structure for input data

### DISCUSSION THREAD

**Discussion by mavijayk** [json-schema-org/community#540] [Q&A]
Hi All, below is the json input file structure I'm receiving daily,
 **ETL** : Informatica Cloud Intelligence
**Source** : Json event files

**sample data**
```json
[
{
	"apple-2023-11-21": [
		{
			"name": "apple",
			"type": "fruit",
			"taste": "sweet"
		}
	],
	"potato-2023-11-21": [
		{
			"name": "potato",
			"type": "veggie",
			"taste": "sour"
		}
	]
}
]
```

**schema I have generated**

```json
{
	"$id": "https://example.com/object1700667140.json", 
	"type": "array",
	"default": [],
	"items":{
		"type": "object",
		"properties": {
			"apple-2023-11-21": {
				"type": "array",
				"default": [],
				"items":{
					"type": "object",
					"properties": {
						"name": {
							"type": "string",
							"default": "",
							"pattern": "^.*$"
						},
						"type": {
							"type": "string",
							"default": "",
							"pattern": "^.*$"
						},
						"taste": {
							"type": "string",
							"default": "",
							"pattern": "^.*$"
				}
		}}},
		"potato-2023-11-21": {
				"type": "array",
				"default": [],
				"items":{
					"type": "object",
					"properties": {
						"name": {
							"type": "string",
							"default": "",
							"pattern": "^.*$"
						},
						"type": {
							"type": "string",
							"default": "",
							"pattern": "^.*$"
						},
						"taste": {
							"type": "string",
							"default": "",
							"pattern": "^.*$"
						}	}	}	}	}}}
```

Above tile name is dynamic value changes in a pattern "fruitname-yyyy-mm-dd" now problem here is, Informatica NOT able to parse the json file which is coming in pattern like above. I tired many options/properties etc. but didn't work. kindly do the needful and appreciate your help.

**Regards**,
vijay Kumar

**Comment by benjagm** [comment]
Hi @mavijayk. If your Pattern for Strings is expected to accept strings you'll need to change it. The current value set to ^.$  allows only a single character string. If you want to allow any string, you can use ^.*$ instead.

**Reply by mavijayk** [reply]
I tried this "^.*$" also but IICS(etl) not even reading data and populating nulls

![IMG_20231123_074502.jpg](https://github.com/json-schema-org/community/assets/56264884/7bbb89ee-6131-482e-8b18-51e5089fdb6d)





![IMG_20231123_074612.jpg](https://github.com/json-schema-org/community/assets/56264884/63cdd6eb-4e9a-4565-8525-134f63699c33)

**Comment by jdesrosiers** [comment]
You'll want to use [`patternProperties`](https://json-schema.org/understanding-json-schema/reference/object#patternProperties) to express property names that fit a pattern.

**Reply by mavijayk** [reply]
If I give patterPropertiesin schema informatica not recognizing attributes

![IMG_20231123_071615.jpg](https://github.com/json-schema-org/community/assets/56264884/350c9a77-cbd7-49eb-80e2-ae869012ec28)




![IMG_20231123_071417.jpg](https://github.com/json-schema-org/community/assets/56264884/b4170bd0-d954-4031-beb0-9b70182f14ea)

Any other suggestions please.

**Reply by jdesrosiers** [reply]
Your `patternProperties` regex only allows one character, but other than that your schema looks fine. I'm not familiar with the tool you're using or what you're trying to do with it, but I'd guess that tool just doesn't know how to deal with `patternProperties`.

**Reply by mavijayk** [reply]
Yes @jdesrosiers  my tool can't deal patternproteries.
